File: frontend/src/app/administration/repository/Repository.ts
export class Repository {
  constructor(
    public type?: string,
    public url?: string,
    public name?: string,
    public username?: string
  ) {
    this.type = type ? type : null;
    this.url = url ? url : null;
    this.name = name ? name : null;
    this.username = username ? username : null;
  }
}

File: maven-api/src/main/java/fr/usubelli/maven/Repository.java
package fr.usubelli.maven;

public class Repository {

    private final String id;
    private final String url;
    private final String name;
    private final String username;
    private final String password;

    public Repository(String id, String url, String name, String username, String password) {
        this.id = id;
        this.url = url;
        this.name = name;
        this.username = username;
        this.password = password;
    }

    public String getId() {
        return id;
    }

    public String getUrl() {
        return url;
    }

    public String getName() {
        return name;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

}
